### PR TITLE
DOC: Explain how to use pre-commit

### PR DESCRIPTION
**Pull request by MartinThoma**
This pull request addresses issue #2580

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by MartinThoma** [comment]
> There's a previous section of the document that details the plugins we use, etc: "Coding Conventions". It also needs updating to maybe just list the plugins we use and mention that we use pre-commit to develop locally?

I'm uncertain which level of detail is desirable / desired. Too much detail and people will not read it. Too little detail and they ask for them.

If we set things up neatly, I think it might be possible not to mention flake8 at all. We could have a pre-commit config to check those things automatically locally and the CI pipeline to check if there are issues.

The other extreme would be to go into detail why the different solutions exist and which alternatives contributors have (e.g. executing it themselves - pre-commit is not mandatory)

**Comment by MartinThoma** [comment]
@JoaoRodrigues Do you happen to know why travis fails?

**Comment by peterjc** [comment]
doc8 didn't like your edits, quoting

```
$ doc8 --ignore-path *.rst # edited for brevity
Scanning...
Validating...
CONTRIBUTING.rst:84: D001 Line too long
CONTRIBUTING.rst:85: D001 Line too long
CONTRIBUTING.rst:86: D001 Line too long
========
Total files scanned = 10
Total files ignored = 1
Total accumulated errors = 3
Detailed error counts:
    - doc8.checks.CheckCarriageReturn = 0
    - doc8.checks.CheckIndentationNoTab = 0
    - doc8.checks.CheckMaxLineLength = 3
    - doc8.checks.CheckNewlineEndOfFile = 0
    - doc8.checks.CheckTrailingWhitespace = 0
    - doc8.checks.CheckValidity = 0
```

**Comment by peterjc** [comment]
I agree that once pre-commit is configured nicely, we don't need to give detailed instructions about all the flake8 plugins etc. It can be far briefer, saying that pre-commit will use flake8, various plugins, black, etc.

**Comment by MartinThoma** [comment]
Ok, then I try to make this brief :-)

**Comment by peterjc** [comment]
When I was testing the pre-commit setup, I was pleasantly surprised that it handled all of the dependencies itself. Initially I thought I'd need to do them manually (or via Tox). For ease of installation this is a big win.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -77,6 +77,23 @@ chapter in the Biopython Tutorial for more information on our test framework:
...
 
 
+Local Development
+-----------------
+
+To simplify the contribution, we provide a `pre-commit
+<https://pre-commit.com/>`_ configuration. Pre-commit is a Python package which
+automatically hooks into git. When you run "git commit" within the biopython
+repository, it will automatically run various fast checks and automatically
+apply simple formatting changes such as black before the commit happens. In
```
Comment (peterjc) [inline CONTRIBUTING.rst:87]: I thought it just checked black style, not applied it?

Comment (peterjc) [reply CONTRIBUTING.rst:87]: How about:

> ... it will automatically run various fast checks (including flake8 and black) before the commit happens.

**Comment by JoaoRodrigues** [comment]
>I was pleasantly surprised that it handled all of the dependencies itself. Initially I thought I'd need to do them manually (or via Tox). For ease of installation this is a big win.

Right? It's very simple indeed. It does hide the dependencies a bit in the pre-commit config file but I think as long as we use it for style checking only, it's wonderful.

I agree with keeping it simple in terms of text!

**Comment by MartinThoma** [comment]
(FYI: I'm not sure if I can do it today; I'm on vacation and my girlfriend gets angry if I spend too much time on the computer :sweat_smile:  )

**Comment by JoaoRodrigues** [comment]
I have a couple of days before I'm in that situation, so you go enjoy and I'll trim the text ;)

**Comment by JoaoRodrigues** [comment]
Shortened the conventions section and addressed Peter's suggestion in the local development section.

**Comment by peterjc** [comment]
Applied, thank you both. And enjoy your vacations too 🎉

**Comment by JoaoRodrigues** [comment]
Interesting - github did not close the PR after merging. Will do it manually.

**Pull request merged**